### PR TITLE
[SPARK-25534][SQL] Make `SQLHelper` trait

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
## What changes were proposed in this pull request?

Currently, Spark has 7 `withTempPath` and 6 `withSQLConf` functions. This PR aims to remove duplicated and inconsistent code and reduce them to the following meaningful implementations.

**withTempPath**
- `SQLHelper.withTempPath`: The one which was used in `SQLTestUtils`.

**withSQLConf**
- `SQLHelper.withSQLConf`: The one which was used in `PlanTest`.
- `ExecutorSideSQLConfSuite.withSQLConf`: The one which doesn't throw `AnalysisException` on StaticConf changes.
- `SQLTestUtils.withSQLConf`: The one which overrides intentionally to change the active session.
```scala
  protected override def withSQLConf(pairs: (String, String)*)(f: => Unit): Unit = {
    SparkSession.setActiveSession(spark)
    super.withSQLConf(pairs: _*)(f)
  }
```

## How was this patch tested?

Pass the Jenkins with the existing tests.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
Could you review this PR, @cloud-fan , @gatorsmile , and @wangyum ?

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/plans/PlanTest.scala [hunk 1]
```diff
@@ -57,7 +57,7 @@ trait CodegenInterpretedPlanTest extends PlanTest {
  * Provides helper methods for comparing plans, but without the overhead of
  * mandating a FunSuite.
  */
-trait PlanTestBase extends PredicateHelper { self: Suite =>
+trait PlanTestBase extends PredicateHelper with SupportWithSQLConf { self: Suite =>
```
Comment (dongjoon-hyun) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/plans/PlanTest.scala]: I put `SupportWithSQLConf.scala` in the same package with `PlanTest`. We can move this to another better place.

**Comment by cloud-fan** [comment]
We have duplicated `withTempPath`, `withTempTable` too. Shall we pull them out as well? The trait name can be `SQLHelper`

**Comment by dongjoon-hyun** [comment]
+1, Sure, @cloud-fan !

**Comment by dongjoon-hyun** [comment]
It turns out that `withTempTable` cannot be here together because it requires `spark.catalog`.
```scala
  def withTempTable(tableNames: String*)(f: => Unit): Unit = {
    try f finally tableNames.foreach(spark.catalog.dropTempView)
  }
```
Since it requires function signature changes, shall we postpone `withTempTable` as another PR?

**Comment by cloud-fan** [comment]
sure

**Review by HyukjinKwon — APPROVED** [review]
LGTM if the tests pass

**Review by cloud-fan — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
It passed. Merged to `master`.

**Pull request closed without merging**